crates/rune/src/compile/context.rs

```diff
@@ -610,6 +621,27 @@ impl Context {
         Ok(())
     }
 
+    /// Install an attribute macro and check for duplicates.
+    fn install_attribute_macro(
+        &mut self,
+        module: &Module,
+        m: &ModuleAttributeMacro,
+    ) -> Result<(), ContextError> {
+        let item = module.item.join(&m.item);
+        let hash = Hash::type_hash(&item);
+        self.attribute_macros.insert(hash, m.handler.clone());
+
+        self.install_meta(ContextMeta {
+            hash,
+            item: Some(item),
+            kind: meta::Kind::Macro,
```

Comment:
This should use a new kind, `meta::Kind::Macro` is expected to be looked up as function-style macros, like `hello!(..)`.

We might also need to do some conflict resolution in case an attribute is named the same as a function-style macro. I.e. in [call contexts](https://github.com/rune-rs/rune/blob/e5b7a3a1095768b26fa05ec4e2857aba78a019f7/crates/rune/src/compile/v1/assemble.rs#L2773) we should prefer to use the latter since you can't *call* an attribute.

You should add a test case to see the issue:
* Register an attribute and a regular macro no the same path, both called `conflict`.
* Try to build an expression like `conflict!()`.